### ISSUE TITLE
need hint on .css and .js

### ISSUE THREAD

**Issue by zahodinikola** [zahodinikola/copy#7]
Hi, here the changes i've made. Pages still looks a bit clumsy, but, as for me, closer to moquap. I have some questions to work on. Here is one of them, i couldn't yet cope with it.
![page1](https://cloud.githubusercontent.com/assets/25704507/23243689/2dd473ec-f989-11e6-8685-9a16357e46d0.png)
On the second picture numbers represent series of questions:
1 - I put text in the &lt;img&gt; element, but can't align it vertically (same one, like on previous picture);
2, 4, 5 - I have to discover, how to transfer data from &lt;input&gt; element to compute with JavaScript code and do some action, depending on results;
3 - I didn't find yet how to arrange some elements's relative position (i wanted to place button in the center, but oops..)
![page2](https://cloud.githubusercontent.com/assets/25704507/23243697/40d2ccbe-f989-11e6-88d4-5c13426a384f.png)
And one about VSCode interface.
![vscode_marks](https://cloud.githubusercontent.com/assets/25704507/23244636/69b9f84a-f98f-11e6-9be7-b484d4af2e48.png)
I am sure, they pointing on something very important. And i hope that i didn't conceal something momentous on the picture with those notes. 
Thank you in advance.

**Comment by humanier** [comment]
Regarding the first page. I think we should consider it is done. Yay!

Do not worry about centering vertically. This not a trivial task (surprise! :)) contrary to horizontal centering. You can read[ this article](http://vanseodesign.com/css/vertical-centering/) on this topic if you want. But I don't think we should spend too much time on this.

**Comment by humanier** [comment]
About second page.
1. Do not worry about vertical centering for now. It looks good enough as for me. If you still feel like working on this - refer to the article from my previous comment.
2. (4,5) - I will try to send you some hints on that tomorrow.
3. The trick here is to shift focus from the button (<input>) itself and think about its parent. You can center it by wrapping it in a new div like that: 
```html
<div style="text-align:center">
      <input ... />
</div>
```

**Comment by zahodinikola** [comment]
Hi, i"ve discovered meaning of the colored marks. You don"t have to answer this question.

**Comment by humanier** [comment]
Good to know you found this solution yourself! By the way there is a nice extension also related to working with Git from VS code: donjayamanne.githistory. Search it in extensions. It adds ability to view Git log (history) inside VS Code.

How to invoke it (from its description):

> Open the file to view the history, and then Press F1 and select/type "Git: View History (git log)", "Git: View File History" or "Git: View Line History".

**Comment by humanier** [comment]
Another useful extension - https://marketplace.visualstudio.com/items?itemName=qinjia.view-in-browser

**Comment by humanier** [comment]
Now about validation.

There are many ways to do that. The most basic one is to use HTML5 validation tags built-in into modern browsers.

1) https://www.w3schools.com/html/html_form_attributes.asp
This can be as simple as adding `... maxlength="10" required>` to the `<input>` tag. That will limit user to 10 characters and will show a tooltip when user clicks "Submit" button without entering any text in the field.

Try that to see how it works but then we need to try second approach.

2) Using pure JavaScript and DOM.
The approach here is to define events handlers for our web page elements. You can read about events [here](https://www.w3schools.com/js/js_events.asp).

 We have two tasks: validate input when "Submit" is clicked and update characters counter as user types. 

Validation can be done by hooking up to onsubmit event of "Submit" button see example [here](https://www.w3schools.com/js/js_validation.asp). In order to hide/show asterisk before the input field name "Review" you can use approach from this [answer ](http://stackoverflow.com/questions/6242976/javascript-hide-show-element).

Updating characters counter can be done by hooking up to **oninput** event of your text input element. See example [here](https://www.w3schools.com/jsref/event_oninput.asp).

Please let me know if you have additional questions (. Regarding this assignment.

I hope we can wrap it up soon and move to the next topic. I am going to spend some time to dive deeper into JavaScript. I found a great  book online with examples and tasks after each chapter. The book is in Russian but that does not matter too much because they usually mention original names of things they talk about. We should make sure you're comfortable with JS before moving on.

**Comment by zahodinikola** [comment]
Hi.
   Until now i did those like looking into CSS and JavaScript. Regerding
CSS i read "CSS - Каскадные таблицы стилей. Подробное руководство. 3-е
издание. Эрик А. Майер" to understand how it handles the elements.
Concerning JavaScript i read
"Devid_Flenagan_-_JavaScript_Podrobnoe_rukovodstvo" and watching tutorial
https://app.pluralsight.com/player?course=javascript-from-scratch&author=jesse-liberty&name=javascript-from-scratch-m5&clip=1&mode=live
.
   Basing on latter i tried to write code for counting symbols are left for
entering, but unfortunatly didn`t cope yet. The main problem is that
message don`t refresh automatically after every char entered, only after i
have pressed enter and only for a second and then returns to it`s initial
value.
  I will workout validation and i guess during this obtain understanding.

2017-02-27 19:50 GMT+02:00 humanier <notifications@github.com>:

**Comment by humanier** [comment]
Regarding JS not refreshing page automatically see my previous comment using different event should - **oninput**

**Comment by humanier** [comment]
I will also add couple code review comments in the JS file itself.

**Comment by zahodinikola** [comment]
start working on validation

**Comment by humanier** [comment]
all looks good except one last thing. 

I will create another issue for it.
